### PR TITLE
Binary search return last value

### PR DESCRIPTION
**Pull request by robertpeacock22**

### CODE REVIEW

File: lib/kitchen_sink/algorithms.ex [hunk 1]
```diff
@@ -87,35 +63,71 @@ defmodule KitchenSink.Algorithms do
...
+  # Strategies
+  defp binary_search_midpoint(start, finish) do
+    start + div(finish - start, 2)
+  end
+
+  defp binary_search_interval(start, finish) do
+    start + (finish - start) / 2.0
+  end
+
+  # Increment bounds
```
Comment (mikestok) [inline lib/kitchen_sink/algorithms.ex]: Is this comment needed?

File: lib/kitchen_sink/algorithms.ex [hunk 2]
```diff
@@ -87,31 +63,67 @@ defmodule KitchenSink.Algorithms do
...
+  defp bounded_increment(to_increment, bound) when to_increment >= bound do
```
Comment (mikestok) [inline lib/kitchen_sink/algorithms.ex]: Opinions sought... For "trivial" functions I have seen and liked the style

    defp bounded_increment(to_increment, bound) when to_increment >= bound, do: bound
    defp bounded_increment(to_increment, _), do: to_increment + 1

**Review by mikestok — APPROVED** [review]

File: lib/kitchen_sink/algorithms.ex (hunk 2 above)
Comment (robertpeacock22) [reply lib/kitchen_sink/algorithms.ex]: To my eye, this pushes the interesting part of the function out to the right, and makes it harder to vertically scan and understand the difference between these functions. Team, other opinions?

File: lib/kitchen_sink/algorithms.ex (hunk 1 above)
Comment (robertpeacock22) [reply lib/kitchen_sink/algorithms.ex]: It was at 1 AM 😸 I like comments that group functions so you can vertically scoot through a file quicker. I find that we are far too anaemic on comments across the board. But not a hill I'm about to die on.

**Review by olafura — APPROVED** [review]

File: lib/kitchen_sink/algorithms.ex (hunk 2 above)
Comment (maedhr) [reply lib/kitchen_sink/algorithms.ex]: I am with @mikestok on this. I find it allows the function definitions to be scanned more easily. Assuming it doesn't push too far out of the width bounds of course.

**Comment by mikestok** [comment]
Increment bounds could be "modify bounds" then to include the decermenters
logically...

Don't  mind the comment for grouping stuff...

**Review by maedhr — APPROVED** [review]

**Review by olafura — APPROVED** [review]

**Pull request merged**